File: app/src/test/java/com/example/radle/todo_calendar2/calendarView/tools/HourTextFormatterTest.java
package com.example.radle.todo_calendar2.calendarView.tools;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class HourTextFormatterTest {

    @Test
    public void format_shouldAddZeroZero_whenTwoDigitsRealHourIsPassed() throws HourTextFormatter
            .NotRealHourNumberException {
        assertEquals("23:00", HourTextFormatter.format(23));
    }

    @Test
    public void format_shouldAddPrefixZeroAndZeroZero_whenOneDigitHourIsPassed() throws
            HourTextFormatter.NotRealHourNumberException {
        assertEquals("00:00", HourTextFormatter.format((0)));
    }

    @Test(expected = HourTextFormatter.NotRealHourNumberException.class)
    public void format_shouldThrowNotRealHourNumberException_whenNumberHigherThan23IsPassed()
            throws HourTextFormatter.NotRealHourNumberException {
        HourTextFormatter.format(24);
    }

    @Test(expected = HourTextFormatter.NotRealHourNumberException.class)
    public void format_shouldThrowNotRealHourNumberException_whenNumberLowerThanZero() throws
            HourTextFormatter.NotRealHourNumberException {
        HourTextFormatter.format(-1);

    }

}

File: app/src/test/java/com/example/radle/todo_calendar2/calendarView/tools/HourTextFormatterTest.kt
package com.example.radle.todo_calendar2.calendarView.tools

import org.junit.Assert.assertEquals
import org.junit.Test

class HourTextFormatterTest {

    @Test
    @Throws(HourTextFormatter.NotRealHourNumberException::class)
    fun format_shouldAddZeroZero_whenTwoDigitsRealHourIsPassed() {
        assertEquals("23:00", hourTextFormatter.format(23))
    }

    @Test
    @Throws(HourTextFormatter.NotRealHourNumberException::class)
    fun format_shouldAddPrefixZeroAndZeroZero_whenOneDigitHourIsPassed() {
        assertEquals("00:00", hourTextFormatter.format(0))
    }

    @Test(expected = HourTextFormatter.NotRealHourNumberException::class)
    @Throws(HourTextFormatter.NotRealHourNumberException::class)
    fun format_shouldThrowNotRealHourNumberException_whenNumberHigherThan23IsPassed() {
        hourTextFormatter.format(24)
    }

    @Test(expected = HourTextFormatter.NotRealHourNumberException::class)
    @Throws(HourTextFormatter.NotRealHourNumberException::class)
    fun format_shouldThrowNotRealHourNumberException_whenNumberLowerThanZero() {
        hourTextFormatter.format(-1)

    }

    companion object {
        private val hourTextFormatter = HourTextFormatter()
    }
}